src/WinRT.Runtime/Projections/SystemTypes.cs

```diff
@@ -55,21 +55,26 @@ public struct DateTimeOffset
         // 'System.DateTimeOffset.Ticks' is relative to 01/01/0001
         public long UniversalTime;
 
+        // Numer of ticks counted between 0001-01-01, 00:00:00 and 1601-01-01, 00:00:00.
+        // You can get this through:  (new DateTimeOffset(1601, 1, 1, 0, 0, 1, TimeSpan.Zero)).Ticks;
+        private const long ManagedUtcTicksAtNativeZero = 504911232000000000;
+        // DO NOT use ToFileTime/FromFileTime, which don't support negative UniversalTime.
+
         public struct Marshaler
         {
             public DateTimeOffset __abi;
         }
 
         public static Marshaler CreateMarshaler(global::System.DateTimeOffset value)
         {
-            return new Marshaler { __abi = new DateTimeOffset { UniversalTime = value.ToFileTime() } };
+            return new Marshaler { __abi = new DateTimeOffset { UniversalTime = value.Ticks - ManagedUtcTicksAtNativeZero } };
         }
 
         public static DateTimeOffset GetAbi(Marshaler m) => m.__abi;
 
         public static global::System.DateTimeOffset FromAbi(DateTimeOffset value)
         {
-            return global::System.DateTimeOffset.FromFileTime(value.UniversalTime);
+            return new global::System.DateTimeOffset(value.UniversalTime + ManagedUtcTicksAtNativeZero, global:: System.TimeSpan.Zero);
```

Comment:
The time we are getting here is going to be UTC.  It looks like .NET core previously then converted it to local time as seen in the link from the issue and threw an exception for any invalid local times.  We probably should do something similar, but it looks like they used [internal APIs](https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/System.Private.CoreLib/src/System/DateTimeOffset.cs#L631) to achieve it.  We can probably achieve something similar by doing something like

```
TimeSpan offset = TimeZoneInfo.Local.GetUtcOffset(utcDateTime);
long localTicks = utcDateTime.Ticks + offset.Ticks;
if (localTicks < DateTime.MinTicks || localTicks > DateTime.MaxTicks)
{
    throw new ArgumentOutOfRangeException();
}

return new global::System.DateTimeOffset(localTicks, offset);
```